src/software/ai/navigator/path_planner/bang_bang_trajectory_1d.cpp

```diff
@@ -0,0 +1,310 @@
+#include "software/ai/navigator/path_planner/bang_bang_trajectory_1d.h"
+
+#include <cmath>
+
+#include "software/geom/algorithms/is_in_range.h"
+#include "software/logger/logger.h"
+
+BangBangTrajectory1D::BangBangTrajectory1D()
+{
+    // Add a default trajectory part to avoid segfaults
+    // when the getters are called before generate
+    addTrajectoryPart(TrajectoryPart());
+}
+
+void BangBangTrajectory1D::generate(double initial_pos, double final_pos,
+                                    double initial_vel, double max_vel, double max_accel,
+                                    double max_decel)
+{
+    // We will overwrite the previous trajectory
+    num_trajectory_parts = 0;
+
+    CHECK(max_vel != 0) << "Max velocity cannot be 0";
+    CHECK(max_accel != 0) << "Max acceleration cannot be 0";
+    CHECK(max_decel != 0) << "Max deceleration cannot be 0";
```

Comment:
Should this assignment happen before the checks? not sure whats so supposed to happen if bad value given for one of the max's.

Reply:
Which assignment are you referring to? In either way, I believe the `CHECK` crashes the program, so it the exact order shouldn't make much of a difference

Reply:
I think he's referring to num_trajectory_parts which looks a little odd but no big deal here